docs-web/intro/how-nginx-ingress-controller-works.md

```diff
@@ -27,229 +28,240 @@ We assume that the reader is familiar with core Kubernetes concepts, such as Pod
-* The IC registers handlers for every informer. Most of the time, a *Handler* creates an entry for the affected resource in the *Workqueue* where a workqueue element includes the type of the resource and its namespace and name. (See the arrow *3. Puts*)
-* The *Workqueue* always tries to drain itself: if there is an element at the front, the queue will remove the element and send it to the *Controller* by calling a callback function.  (See the arrow *4. Sends*)
-* The *Controller* is the primary component in the IC, which represents the control loop. We will explain the component in the [The Control Loop section](#the-control-loop). For now, it suffices to know that to process a workqueue element, it will get the latest version of the resource from the *Store* (see the arrow *5. Gets*), reconfigure *NGINX* according to the resource (see the arrow *6. Reconfigures*) and update the resource status and emit an event via the *Kubernetes API* (see the arrow  *7. Updates status and emits events*).
+* For every resource type the IC monitors, it creates an [*Informer*](https://pkg.go.dev/k8s.io/client-go@v0.21.0/tools/cache#SharedInformer). The *Informer* includes a *Store* that holds the resources of that type. To keep the *Store* in sync with the latest versions of the resources in the cluster, the *Informer* calls the Watch and List *Kubernetes APIs* for that resource type (see the arrow *1. Watches and Lists* on the diagram).
+* When a change happens in the cluster (for example, a new resource is created), the *Informer* updates its *Store* and invokes [*Handlers*](https://pkg.go.dev/k8s.io/client-go@v0.21.0/tools/cache#ResourceEventHandler) (see the arrow *2. Invokes*) for that *Informer*.
+* The IC registers handlers for every *Informer*. Most of the time, a *Handler* creates an entry for the affected resource in the *Workqueue* where a workqueue element includes the type of the resource and its namespace and name. (See the arrow *3. Puts*.)
+* The *Workqueue* always tries to drain itself: if there is an element at the front, the queue will remove the element and send it to the *Controller* by calling a callback function. (See the arrow *4. Sends*.)
+* The *Controller* is the primary component in the IC, which represents the control loop. We explain the components in [The Control Loop](#the-control-loop) section. For now, it suffices to know that to process a workqueue element, the *Controller* component gets the latest version of the resource from the *Store* (see the arrow *5. Gets*), reconfigures *NGINX* according to the resource (see the arrow *6. Reconfigures*), updates the resource status, and emits an event via the *Kubernetes API* (see the arrow  *7. Updates status and emits events*).
```

Comment:
Please confirm this edit. I wasn't too sure what the "it" in "... it will get the latest version" refers to. I assumed it was the _Controller_ component.

Reply:
I confirm 👍  it is the Controller component